cmd/controller/app/database/mongodb/compute.go

```
@@ -17,19 +17,87 @@
 package mongodb
 
 import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.uber.org/zap"
+
 	"github.com/cisco-open/flame/pkg/openapi"
 )
 
 // RegisterCompute creates a new cluster compute specification and returns ComputeStatus
 func (db *MongoService) RegisterCompute(computeSpec openapi.ComputeSpec) (openapi.ComputeStatus, error) {
-	// TODO implement the method
+	// First check if the compute was previously registered
+	filter := bson.M{"computeid": computeSpec.ComputeId}
+	checkResult := db.computeCollection.FindOne(context.TODO(), filter)
```

What happens if two deployers with the same id turn up?

Should apikey be used to check if one of them can't register the same id?

if the user-provided apikey matches one in the database, you can return nil. otherwise, you should return an error saying that "the compute id is already in use".